plugin/route53/route53.go

```diff
@@ -140,10 +143,79 @@ func (h *Route53) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	return dns.RcodeSuccess, nil
 }
 
+const escapeSeq = `\\`
+
+// maybeUnescape parses s and converts escaped ASCII codepoints (in octal) back
+// to its ASCII representation.
+//
+// From AWS docs:
+//
+// "If the domain name includes any characters other than a to z, 0 to 9, -
+// (hyphen), or _ (underscore), Route 53 API actions return the characters as
+// escape codes."
+//
+// For our purposes (and with respect to RFC 1035), we'll fish for a-z, 0-9,
+// '-', '.' and '*' as the leftmost character (for wildcards) and throw error
+// for everything else.
+//
+// Example:
+//   `\\052.example.com.` -> `*.example.com`
+//   `\\137.example.com.` -> error ('_' is not valid)
+func maybeUnescape(s string) (string, error) {
```

Comment:
you should be able to lift the code out from the dns/msg.go in the packDomainName function which does something similar, but works on a []byte.

Reply:
Hm I took a quick pass - seems not a ton I could use as-is but `dddStringToByte` is kind of cool though - I think I'm going to plug modified version (`oooStringTobyte`) in place of `ParseInt`. Lmk.

Reply:
ok, we may be able to make something more generic and useful in miekg/dns